File: ReptileMorphCalculator.java
import java.util.*;

class ReptileMorphCalculator {
    public static void main(String[] args) {
        reptileMorphCalculator();
    }

    public static void reptileMorphCalculator() {
        Scanner scanner = new Scanner(System.in);

        System.out.println("\nKyle.Exotic Reptile Morph Calculator");
        System.out.println("\nThis Tool can help the Reptile World to calculate the potential offspring from parents reptile.\n");

        String species = "";
        Map<String, Double> morphPrices = new HashMap<>();
        List<String> possibleMorphs = new ArrayList<>();
        double basePrice = 0.0;

        System.out.print("Enter the reptile species (e.g., Ball Python, Leopard Gecko, Reticulated Python): ");
        species = scanner.nextLine();

        if (species.toLowerCase().contains("ball python")) {
            possibleMorphs = Arrays.asList("Normal", "Albino", "Piebald", "Pastel", "Spider", "Clown", "Banana");
            basePrice = 1000.0;
        } else if (species.toLowerCase().contains("leopard gecko")) {
            possibleMorphs = Arrays.asList("Normal", "Tangerine", "Hypo", "Super Hypo", "Albino", "Blizzard");
            basePrice = 2000.0;
        } else if (species.toLowerCase().contains("reticulated python")) {
            possibleMorphs = Arrays.asList("Albino", "Tiger", "Golden Child", "Motley");
            basePrice = 20000.0;
        } else {
            possibleMorphs = Arrays.asList("Normal", "Albino", "Patternless", "Wild Type");
            basePrice = 1500.0;
        }

        System.out.println("\nAvailable morphs for " + species + ":");
        for (int i = 0; i < possibleMorphs.size(); i++) {
            String morph = possibleMorphs.get(i);
            System.out.println((i + 1) + ". " + morph);
            morphPrices.put(morph, basePrice * (1 + ((i + 1) * 0.5)));
        }

        String parent1 = "";
        String parent2 = "";

        while (!possibleMorphs.contains(parent1)) {
            System.out.print("\nEnter morph for Parent 1: ");
            parent1 = scanner.nextLine();
            if (!possibleMorphs.contains(parent1)) {
                System.out.println("Invalid morph. Please choose from the list above.");
            }
        }

        while (!possibleMorphs.contains(parent2)) {
            System.out.print("Enter morph for Parent 2: ");
            parent2 = scanner.nextLine();
            if (!possibleMorphs.contains(parent2)) {
                System.out.println("Invalid morph. Please choose from the list above.");
            }
        }

        System.out.println("\nCalculating possible offspring morphs...Please wait Hehe!");

        Set<String> offspringMorphs = new HashSet<>();

        if (parent1.equals(parent2)) {
            offspringMorphs.add(parent1);
            if (!parent1.equals("Normal")) {
                offspringMorphs.add("Normal");
            }
        } else {
            offspringMorphs.add(parent1);
            offspringMorphs.add(parent2);
            offspringMorphs.add("Normal");
        }

        // Special combo ni Kyle mag breed hehe.
        if ((parent1.equals("Albino") || parent2.equals("Albino")) &&
                (parent1.equals("Piebald") || parent2.equals("Piebald")) &&
                species.toLowerCase().contains("ball python")) {
            offspringMorphs.add("Albino Piebald");
        }

        // Display results
        System.out.println("\nPossible Offspring Morphs:");
        List<String> sortedMorphs = new ArrayList<>(offspringMorphs);
        Collections.sort(sortedMorphs);

        for (String morph : sortedMorphs) {
            double price = morphPrices.getOrDefault(morph, basePrice);
            System.out.printf("- %s (Estimated value: ₱%.2f)%n", morph, price);
        }

        // Calculate average value
        if (!offspringMorphs.isEmpty()) {
            double totalValue = 0.0;
            for (String morph : offspringMorphs) {
                totalValue += morphPrices.getOrDefault(morph, basePrice);
            }
            double avgValue = totalValue / offspringMorphs.size();
            System.out.printf("%nAverage offspring value: ₱%.2f%n", avgValue);
        }

        System.out.println("\nThank you for using Kyle.Exotic Reptile Morph Calculator!");
        scanner.close();
    }
}

File: ReptileMorphCalculator.py
# Kyle Bryan David ITCC121 Project
def reptile_morph_calculator(): #Defining Main Functions 
    
    print("Kyle.Exotic Reptile Morph Calculator")
    print("This Tool can help the Reptile World to calculate the potential offspring from parents reptile.\n")
    
    
    species = ""
    morph_prices = {}  
    possible_morphs = []  
    
    
    species = input("Enter the reptile species (e.g., Ball Python, Leopard Gecko, Reticulated Python): ")
    
    
    if "ball python" in species.lower():
        possible_morphs = ["Normal", "Albino", "Piebald", "Pastel", "Spider", "Clown", "Banana"]
        base_price = 1000.0  
    elif "Leopard gecko" in species.lower():
        possible_morphs = ["Normal", "Tangerine", "Hypo", "Super Hypo", "Albino", "Blizzard"]
        base_price = 2000.0
    elif "reticulated python" in species.lower():
        possible_morphs = ["Albino","Tiger","Golden Child","Motley"]
        base_price = 20000.0
    else:
        possible_morphs = ["Normal", "Albino", "Patternless", "Wild Type"]
        base_price = 1500.0
    
    
    print(f"\nAvailable morphs for {species}:")
    for i, morph in enumerate(possible_morphs, 1):
        print(f"{i}. {morph}")
        
        morph_prices[morph] = base_price * (1 + (i * 0.5))  
    
    
    parent1 = ""
    parent2 = ""
    
    while parent1 not in possible_morphs:
        parent1 = input("\nEnter morph for Parent 1: ")
        if parent1 not in possible_morphs:
            print("Invalid morph. Please choose from the list above.")
    
    while parent2 not in possible_morphs:
        parent2 = input("Enter morph for Parent 2: ")
        if parent2 not in possible_morphs:
            print("Invalid morph. Please choose from the list above.")
    
   
    print("\nCalculating possible offspring morphs...Please wait Hehe!")
    
    
    
    offspring_morphs = set()  
    
    
    if parent1 == parent2:
        offspring_morphs.add(parent1)
        if parent1 != "Normal":
            offspring_morphs.add("Normal")  
    else:
        offspring_morphs.add(parent1)
        offspring_morphs.add(parent2)
        offspring_morphs.add("Normal")  
    
     # Special combo mag breed si kyle
    if ("Albino" in [parent1, parent2]) and ("Piebald" in [parent1, parent2]) and "Ball Python" in species:
        offspring_morphs.add("Albino Piebald")  
    
    # Display results
    print("\nPossible Offspring Morphs ")
    for morph in sorted(offspring_morphs):
        price = morph_prices.get(morph, base_price)
        print(f"- {morph} (Estimated value: ₱{price:.2f})")
    
   # Calculate average value
    if offspring_morphs:
        avg_value = sum(morph_prices.get(m, base_price) for m in offspring_morphs) / len(offspring_morphs)
        print(f"\nAverage offspring value: ₱{avg_value:.2f}")
    
        
    
    print("\nThank you for using Kyle.Exotic Reptile Morph Calculator!")

# Run the program
if __name__ == "__main__":
    reptile_morph_calculator()